### PR TITLE
LPD-30650 Allow Workflow admin to see date and user information on history

### PR DESCRIPTION
**Pull request by danvb112**
Related Issues:

https://liferay.atlassian.net/browse/LPD-30650
https://liferay.atlassian.net/browse/LPD-30897
https://liferay.atlassian.net/browse/LPD-30949

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:playwright

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -243,7 +248,7 @@ components:
                     items:
                         $ref: "#/components/schemas/Transition"
                     readOnly: true
-                    type: array
+                    type: arrayw
```
Comment (marcelabc) [inline modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml:251]: @vngoliveira Is this a typing error?

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -211,6 +211,11 @@ components:
                 content:
                     type: string
                     x-json-string: true
+                creator:
+                    $ref: "#/components/schemas/Creator"
+                    description:
+                        The person who created the workflow
```
Comment (marcelabc) [inline modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml:217]: Change the description to:  `The workflow's creator`.

Comment (marcelabc) [reply modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml:217]: @vngoliveira, This matches the existing descriptions of the Rest APIs from the portal.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @vngoliveira @danvb112 
I finished my review on the backend part. Let me know if you have questions :smile:

**Comment by marcelabc** [comment]
Please, rename the commit `LPD-30949 autogenerated (buildRest)` to `LPD-30949 buildRest`

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml (hunk 1 above)
Comment (danvb112) [reply modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml]: Done!

File: modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml (hunk 2 above)
Comment (danvb112) [reply modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/rest-openapi.yaml]: Thanks @marcelabc!

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
Hey @marcelabc I made all the changes you requested, could you please double-check? Thanks for the review! :smile:

**Review by marcelabc — APPROVED** [review]
LGTM 🎉
(backend part)

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:workflow

**Comment by marcelabc** [comment]
(running CI to anticipate results)

**Comment by danvb112** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
I found two inconsistencies:

I need to carry out more tests to determine whether these failures are due to the changes made to these PRs. 

The first inconsistency noticed is due to the rendering of the tabs, they are being rendered one below the other.

Furthermore, when we reload the page, we lose the version references and are left with just one version. It is necessary to go back to the workflow definitions page and open the JSP again, this way we have our versions list restored.

**Comment by igor-franca** [comment]
I'm adding the `Author Action Required` label just to signal that this PR needs changes before moving forward. I know Daniel is on vacation, I should work on that in the next days.

**Comment by marcelabc** [comment]
Thank you for your review @igor-franca !

**Comment by igor-franca** [comment]
Hey @marcelabc, updates:

In relation to the errors pointed out previously.

The first error also occurs without Daniel's modifications, a tab rendering problem. I don't know if it occurs in all resolutions.

The second reported problem  does not occur on Daniel's branch using FF, only on Daniel's branch without using FF and on master. So I think we can open a common bug here and add it to the team flow. Note that the tabs are not side by side in the video.

On the other hand, I identified an inconsistency related to the modification time in the history:

https://github.com/user-attachments/assets/96570dd9-4231-41b4-9d7d-08aed31d2f2d

**Comment by marcelabc** [comment]
Hey @igor-franca 
Thank you for the comment and video.

In summary, you found 3 issues:
1. This one happens on master (without this code).
   > We can open a bug and ask the PO to define the FP.
2. This happens with this code but only with the FF on.
   > We can open a bug and ask the PO to define the FP. They can decide if it is an impedibug for the story or not.
3. The inconsistency with the time zone. 
   > We can check how this works in 7.3.x (before the migration). On 7.3.x, this feature exists, and we should focus on keeping the same behavior and not creating regression. Today, we don't have this behavior on master, but we should replicate the behavior of 7.3.x.

Could we move on with the review (run the CI, send it to Brian) while we open the bugs and investigate the other scenarios, or do you think there is code refactoring to do?

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/DefinitionInfo/DetailsTab.tsx [hunk 3]
```diff
@@ -37,30 +39,36 @@ export function DetailsTab({definitionInfo}: DetailsTabProps) {
...
 					{titleTotalModifications.toUpperCase()}
 				</label>
 
-				<span>{revisionMessage}</span>
+				<span className="lfr-workflow__details-tab-info-details">
+					{revisionMessage}
+				</span>
 			</div>
-		</>
+		</div>
```
Comment (igor-franca) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/DefinitionInfo/DetailsTab.tsx:72]: I see that we can improve this component, this code snippet is repeated tree times:

```
	<div className="lfr-workflow__details-tab-info-container">
		<label className="lfr-workflow__details-tab-info-title">
			{x}
		</label>

		<span className="lfr-workflow__details-tab-info-details">
			{y}
		</span>
	</div>
```

We can create an array to store the constants declared above and loop through the array to render these contents

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 4]
```diff
@@ -232,20 +217,25 @@ export default function UpperToolbar({
...
-			active,
-			content: xmlDefinition,
-			name,
-			title: definitionTitle,
-			title_i18n: definitionTitleTranslations,
-			version,
-		});
+		const publishedOrSavedDefinitionResponse =
+			await saveOrPublishDefinitionRequest({
+				active,
```
Comment (igor-franca) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:222]: I see `publishedOrSavedDefinition` and `saveOrPublishDefinition`. 

I believe we should standardize the order of terms.

**Review by igor-franca — COMMENTED** [review]
Hey Marcela, I will make these changes now and continue with the testing routines.

**Comment by igor-franca** [comment]
> Hey @igor-franca Thank you for the comment and video.
> 
> In summary, you found 3 issues:
> 
> 1. This one happens on master (without this code).
>    > We can open a bug and ask the PO to define the FP.
> 2. This happens with this code but only with the FF on.
>    > We can open a bug and ask the PO to define the FP. They can decide if it is an impedibug for the story or not.
> 3. The inconsistency with the time zone.
>    > We can check how this works in 7.3.x (before the migration). On 7.3.x, this feature exists, and we should focus on keeping the same behavior and not creating regression. Today, we don't have this behavior on master, but we should replicate the behavior of 7.3.x.
> 
> Could we move on with the review (run the CI, send it to Brian) while we open the bugs and investigate the other scenarios, or do you think there is code refactoring to do?

This happens with this code but only with the FF ~on~ off. So I think we can just
> We can open a bug and ask the PO to define the FP.

In short, the only problem that occurs only in this branch is the time zone.

**Comment by igor-franca** [comment]
I invested some time trying to resolve the inconsistency in the schedule, but I was unsuccessful.

**Pull request closed without merging**